subsys/zigbee/osif/zb_nrf_serial.c

```diff
@@ -238,6 +242,10 @@ void zb_osif_serial_init(void)
 
 	/* Enable rx interrupts. */
 	uart_irq_rx_enable(uart_dev);
+
+#if defined(CONFIG_ZBOSS_TRACE_UART_LOGGING) || defined(CONFIG_ZBOSS_TRACE_USB_CDC_LOGGING)
+	zb_logger_init();
```

Comment:
Conceptually, this initialization should not be here.
It would be much cleaner if `zb_logger_init` was called unconditionally, and the backend initialization calls placed inside the `zb_logger_init`, depending on the selection of Kconfig options.